exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -102,3 +139,35 @@ func (amp *AzureMachinePool) SetNetworkInterfacesDefaults() {
 		}
 	}
 }
+
+// findParentMachinePool finds the parent MachinePool for the AzureMachinePool.
+func (amp *AzureMachinePool) findParentMachinePool(cli client.Client) (*expv1.MachinePool, error) {
```

Comment:
did you check if there were any existing helpers in CAPI or CAPZ that do this? If not, seems like a good bit of reusable functionality, might make more sense to place it in controller helpers or utils instead of unexported in the webhook file

Reply:
Looks like there is actually an [existing helper](https://github.com/willie-yao/cluster-api-provider-azure/blob/a8c304ba616fc1c7306179e1e958bf7f6bb89656/controllers/helpers.go#L129) in the controller for getting the owner cluster name. However, I'm unable to import from the controller package from within exp v1beta1 (I'm not sure why). I'll see if there's another way for me to reuse the function. If not, I'll move the function to somewhere reusable!